File: java-code/src/main/java/refactoring/organizing_data/change_reference_to_value/before/Person.java
package refactoring.organizing_data.change_reference_to_value.before;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@AllArgsConstructor
public class Person {
  private TelephoneNumber telephoneNumber;

  public String getOfficeAreaCode() {
    return this.telephoneNumber.getAreaCode();
  }

  public void setOfficeAreaCode(String arg) {
    this.telephoneNumber.setAreaCode(arg);
  }

  public String getOfficeNumber() {
    return this.telephoneNumber.getNumber();
  }

  public void setOfficeNumber(String arg) {
    this.telephoneNumber.setNumber(arg);
  }


}

File: go-code/organizing_data/change_reference_to_value/before/person.go
package before

type Person struct {
	telephoneNumber TelephoneNumber
}

func (p *Person) getOfficeAreaCode() string {
	return p.telephoneNumber.areaCode
}

func (p *Person) setOfficeAreaCode(arg string) {
	p.telephoneNumber.areaCode = arg
}

func (p *Person) getOfficeNumber() string {
	return p.telephoneNumber.number
}

func (p *Person) setOfficeNumber(arg string) {
	p.telephoneNumber.number = arg
}
